### PR TITLE
update virtualservice checksum based on ref names instead of ref checksums

### PR DESCRIPTION
**Pull request by chauhanshubham**

### CODE REVIEW

**Comment by chauhanshubham** [comment]
build ako

**Comment by chauhanshubham** [comment]
build ako

**Comment by sudswasavi** [comment]
build ako

**Review by monotosh-avi — DISMISSED** [review]

**Comment by sudswasavi** [comment]
@chauhanshubham / @monotosh-avi - we can still review this for `master` but it does not have to be cherry-picked to 1.4.1

**Comment by monotosh-avi** [comment]
build ako

**Comment by sudswasavi** [comment]
@akshayhavile / @monotosh-avi can you pls review this PR carefully?

File: internal/nodes/avi_model_nodes.go [hunk 1]
```diff
@@ -619,47 +619,45 @@ func (v *AviVsNode) CalculateCheckSum() {
...
+		checksumStringSlice = append(checksumStringSlice, "HTTPDS"+ds.Name)
 	}
 
 	for _, httppol := range v.HttpPolicyRefs {
-		httppolChecksum += httppol.GetCheckSum()
-	}
-
-	for _, sninode := range v.SniNodes {
-		sniChecksum += sninode.GetCheckSum()
+		checksumStringSlice = append(checksumStringSlice, "HttpPolicy"+httppol.Name)
```
Comment (monotosh-avi) [inline internal/nodes/avi_model_nodes.go:629]: As we are considering the sniNodes in the parent VS checksum anymore, lets check if update in the SNI VS does impact the model checksum. I think it should.

**Review by monotosh-avi — COMMENTED** [review]
Looks good to me, one comment to verify SNI

**Review by akshayhavile — COMMENTED** [review]
+1 : For Monotosh's comment. Also we need to do same changes at evh side.

**Comment by chauhanshubham** [comment]
build ako

**Comment by chauhanshubham** [comment]
build ako

**Review by sudswasavi — APPROVED** [review]

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**